# Técnicas de Sumarización — TitanicAsimetría (Skewness), Curtosis (Kurtosis) y Tablas Cruzadas (Cross-Tabulation)

In [ ]:
import pandas as pdurl = "https://raw.githubusercontent.com/mricardo89/data-mining/main/Unit-II/datasets/titanic.csv"df = pd.read_csv(url)df.head()

(891, 12)

## 1. Asimetría (Skewness) — columna `Age`

In [ ]:
df['Age'].skew()

0.38910778230082704

**Interpretación:** El skew de `Age` es **0.389**, un valor **positivo y moderado** (dentro del rango de asimetría leve/moderada, entre 0 y 0.5-1). Físicamente esto significa que la distribución tiene una **cola ligera hacia la derecha**: la mayoría de los pasajeros eran jóvenes o de mediana edad, y **hubo más jóvenes que ancianos** — pero unos cuantos pasajeros de edad avanzada estiran la cola derecha de la distribución sin llegar a ser un sesgo extremo. No es una asimetría tan fuerte como la de `Fare` o `Installs` en ejercicios anteriores.

## 2. Curtosis (Kurtosis) — columna `Fare`

In [ ]:
df['Fare'].kurt()

33.39814088089868

**Interpretación:** La curtosis de `Fare` es **33.4**, un número **extremadamente alto** (una distribución normal tiene curtosis ≈ 0, usando el exceso de curtosis que calcula pandas). Esto indica una distribución **leptocúrtica**: un pico muy alto y concentrado cerca de valores bajos, junto con **colas mucho más "gordas" o pesadas** de lo normal — es decir, hay una probabilidad mucho mayor de encontrar valores extremos (outliers) de lo que predeciría una distribución normal.Esto **confirma matemáticamente** los outliers de tarifas de 512 libras que se encontraron con el método del IQR en clase: una curtosis tan alta es la "firma numérica" de que existen valores atípicos extremos jalando las colas de la distribución.

## 3. Tablas Cruzadas (Cross-Tabulation) — `Pclass` vs `Survived`

In [ ]:
pd.crosstab(df['Pclass'], df['Survived'])

Survived    0    1Pclass            1          80  1362          97   873         372  119

In [ ]:
pd.crosstab(df['Pclass'], df['Survived'], normalize='index') * 100

Survived          0          1Pclass                        1         37.037037  62.9629632         52.717391  47.2826093         75.763747  24.236253

**Interpretación:** La tabla cruzada muestra un patrón muy claro: **a menor clase social, menor tasa de supervivencia**. En 1ra clase sobrevivió el **62.96%**, en 2da clase el **47.28%**, y en 3ra clase solo el **24.24%**. De los 372 pasajeros de 3ra clase que murieron, es el grupo más numeroso y con peor proporción de sobrevivientes de todo el barco.Esto sugiere que la clase socioeconómica del pasajero (que probablemente determinaba la cercanía del camarote a los botes salvavidas, y posiblemente el orden de prioridad de evacuación) fue un factor decisivo en quién sobrevivió — no solo el género, como se vio en la actividad anterior.